ome_zarr/writer.py

```diff
@@ -126,17 +126,13 @@ def _validate_datasets(
     datasets: List[dict], dims: int, fmt: Format = CurrentFormat()
 ) -> List[Dict]:
 
-    validated_datasets = []
     if datasets is None or len(datasets) == 0:
         raise ValueError("Empty datasets list")
     transformations = []
     for dataset in datasets:
-        if isinstance(dataset, str):
-            validated_datasets.append({"path": dataset})
-        elif isinstance(dataset, dict):
+        if isinstance(dataset, dict):
```

Comment:
This might have been simpler as `assert isinstance(dataset, dict)`

Reply:
Now I'm pondering if there's some generic `assertion` or `validation` module that one could use to skip for loops where something isn't valid without just throwing a `ValueError`.

Reply:
No strong opinion on using `assert` vs conditional and a custom error like `ValueError`. If we collectively agree on the former version for now, I can quickly go over the codebase after this PR.

My mid-term hope is to replace this internal validation code by a generic validation framework based on schemas/constraints. I expect this should also bring typical features like being able to collect all validation errors like the `Validator.iter_errors()` API of the `jsonschema` module.

Reply:
Exactly. No strong opinion on `assert isinstance` especially if we'll be reviewing soon _anyway_.